src/sotauptaneclient.h

```diff
@@ -44,13 +42,14 @@ class SotaUptaneClient {
   Config &config;
   event::Channel *events_channel;
   Uptane::Repository &uptane_repo;
-  const boost::shared_ptr<INvStorage> storage;
-  boost::shared_ptr<PackageManagerInterface> pacman;
+  const std::shared_ptr<INvStorage> storage;
+  std::shared_ptr<PackageManagerInterface> pacman;
   HttpInterface &http;
   int last_targets_version;
   Json::Value operation_result;
   std::unique_ptr<IpUptaneConnection> ip_uptane_connection;
   std::unique_ptr<IpUptaneConnectionSplitter> ip_uptane_splitter;
+  bool shutdown = {false};
```

Comment:
`std::atomic<bool> shutdown{false};` should probably be used for this kind of inter-thread signaling